packages/next-swc/crates/next-api/src/project.rs

```diff
@@ -639,6 +676,24 @@ impl Project {
         for (pathname, page_route) in pages_project.routes().await?.iter() {
             match routes.entry(pathname.clone()) {
                 Entry::Occupied(mut entry) => {
+                    ConflictIssue {
+                        path: self.project_path(),
+                        title: StyledString::Text(
+                            format!("App Router and Pages Router both match path: {}", pathname)
+                                .to_string(),
+                        )
+                        .cell(),
+                        description: StyledString::Text(
+                            "Next.js does not support having both App Router and Pages Router \
+                             routes matching the same path. Please remove one of the conflicting \
+                             routes."
+                                .to_string(),
+                        )
+                        .cell(),
+                        severity: IssueSeverity::Error.cell(),
```

Comment:
Technically you could omit severity and description since they are constant. Just return the fixed value in the issue implementation

Reply:
Yeah wasn't sure if there was a reason this wasn't done in the other issues, copied this one from app_structure.rs DirectoryTreeIssue